nmostesting/GenericTest.py

```diff
@@ -217,17 +217,29 @@ def prepare_CORS(self, method):
         return headers
 
     # 'check' functions return a Boolean pass/fail indicator and a message
-    def check_CORS(self, method, headers):
+    def check_CORS(self, method, headers, expect_methods=None, expect_headers=None):
         """Check the CORS headers returned by an API call"""
         if 'Access-Control-Allow-Origin' not in headers:
-            return False, "Incorrect CORS headers: {}".format(headers)
-        if method == "OPTIONS":
+            return False, "'Access-Control-Allow-Origin' not in CORS headers: {}".format(headers)
+        if method.upper() == "OPTIONS":
             if 'Access-Control-Allow-Headers' not in headers:
-                return False, "Incorrect CORS headers: {}".format(headers)
```

Comment:
I think Access-Control-Allow-Headers is only required if the request had Access-Control-Request-Headers? I.e. perhaps if `expect_headers` is non-empty?